test/dx_test.dart

```diff
@@ -2,13 +2,24 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:dx/dx.dart' as dx;
+import 'dart:io';
+import 'package:path/path.dart' as path;
 import 'package:test/test.dart';
 
+import 'package:dx/src/scanner.dart';
+
 void main() {
-  group('main', () {
-    test('works', () {
-      dx.main();
-    });
+  final languageDirectory = new Directory('test/language');
+  if (!languageDirectory.existsSync()) {
+    throw 'Directory not found: ${languageDirectory.path}';
+  }
+
+  group('language test', () {
```

Comment:
You'll want a lot of tests to make sure the various parts of this are working correctly.  It would be frustrating to find out an error encountered in the Parser is due to a scanner quirk.

Reply:
Yeah, this is just to get the process going. I'm thinking of adding metadata in the test files that contain expectations. Sort of like Dart language tests embed expected analyzer warnings in the test code.